### ISSUE TITLE
Improve querying icons in tests

### ISSUE THREAD

**Issue by nemanjaglumac** [metabase/metabase#13224]
**Is your feature request related to a problem? Please describe.**
There isn't currently a good, universal way to target icons in tests that would be in line with the best practices. We're often forced to target CSS classes, which is the least favorable option.

On the other hand, icons are not accessible (this old thread #1462 recognized the need for similar improvements). I believe that both problems can be improved with the proposed solution.

**Describe the solution you'd like**
- React Testing Library favors querying elements [by their role](https://testing-library.com/docs/guide-which-query). 
- according to [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Role_Img) and every other source I found, SVGs should **(at minimum)** have `role="img"` set, so that screen readers skip reading their child nodes

In many tests we don't actually need to check if some particular icon was rendered. Only that **an** icon is present in the component. `getByRole("img)` takes care of that.

The best part: this requires touching only one file (`icon_paths.js`), while providing huge benefits.

**Describe alternatives you've considered**
- further a11y improvements could require adding `<title>` and `<description>` accompanied with [`aria-lebeledby`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute) attribute
    - that could help pinpoint the exact icon
- @paulrosenzweig [proposed an idea](https://github.com/metabase/metabase/pull/13213#discussion_r484956942) of using a custom made query, which is also something to think about
- bottom line, even if there is still a need to target some icons using regular DOM querySelector, it will be container and very rare 
- btw, `@testing-library/jest-dom/extend-expect` has a few helpfull methods for the situations where we need to assert that the selected icon has indeed a certain class (`toHaveClass()`, `toHaveAttribute()`, etc)

**How important is this feature to you?**
It is a huge blocker while writing tests, yet it requires a minimal intervention. I am preparing a PR if this gets 👍

**Additional context**
Let's keep some helpful resources here:
- https://developer.mozilla.org/en-US/docs/Glossary/AOM
- https://www.w3.org/TR/accname-1.1/
- https://css-tricks.com/accessible-svgs/
- https://developer.paciellogroup.com/blog/2013/12/using-aria-enhance-svg-accessibility/

**Comment by nemanjaglumac** [comment]
@paulrosenzweig - I've been reading a lot of a11y articles today and playing with the voiceover on mac to hear how different combinations of aria attributes sound to a screen reader.
- good news: adding `aria-label="foo icon"` on top of `role="img"` helps pinpoint the icon you want (foo being icon's `name` attribute)
    - `getByRole("img", {name: /foo icon/i})` - it's a bit more verbose but it covers both attributes (adds certainty)
    - `getByLabelText(/foo icon/i)` - works just fine but I'm leaning towards the first query
    - removes the need for `data-testid`
- not so good news: we aren't necessarily doing a huge favor to the potential disabled users. It's kind of a mixed bag, really. 
    - Screen reader will read this as "Foo icon, image", which is better than nothing but not extremely helpful without the proper title and description elements. They depent on the context and would need to be set dynamically. It feels messy and will potentially touch a lot of code.
    - another downside is when icon is accompanied by some text, screen reader shouldn't skip it (and again, it would be challenging to track all use cases and re-write the code for that)

So, our only a11y gain is for the cases where icon is alone, and even then it should probably be wrapped in a div with the `role="button"`. On the other hand, this solves our need to target icons in tests.
It's a bit selfish but I'm feeling ok with that trade-off. Thoughts?

**Comment by paulrosenzweig** [comment]
@nemanjaglumac 

> better than nothing but not extremely helpful without the proper title and description elements

We often have icons wrapped in our <Tooltip> component, which could be a good source for that info. 

>  it should probably be wrapped in a div with the `role="button"`

It'd be really cool to have our component library encourage good a11y patterns. e.g. our `<Button>` component takes an `icon` prop. Maybe we should add `role="button"` and ensure we always use that for clickable icons (adding appropriate styling options).

---

I agree that stuff is beyond the scope of this issue though. Improving testing with a a11y win in some (but not all) cases sounds great to me.
